### PR TITLE
Add GOPATH check

### PR DESCRIPTION
**Pull request by ScorpioCPH**
Fix https://github.com/kubernetes/test-infra/issues/4927

### CODE REVIEW

**Comment by krzyzacy** [comment]
/hold
please answer https://github.com/kubernetes/test-infra/issues/4927#issuecomment-336522289

**Comment by ScorpioCPH** [comment]
@krzyzacy Done https://github.com/kubernetes/test-infra/issues/4927#issuecomment-336762695

PTAL.

File: kubetest/util.go [hunk 1]
```diff
@@ -63,7 +63,12 @@ func init() {
 
 // Returns $GOPATH/src/k8s.io/...
 func k8s(parts ...string) string {
-	p := []string{os.Getenv("GOPATH"), "src", "k8s.io"}
+	gp := os.Getenv("GOPATH")
+	if gp == "" {
+		return "Error: GOPATH not set, please check your golang env setting"
```
Comment (krzyzacy) [inline kubetest/util.go]: okay, make the check returns both `error` and `string` then, don't mix them up, and let the caller check for error

**Comment by ScorpioCPH** [comment]
@krzyzacy Updated, PTAL.

**Comment by BenTheElder** [comment]
Can we just check this once in main instead of checking for a single error in a lot of places?
It seems fine to give a better error to inform the user that `$GOPATH` is required but I don't think we need to clutter the rest of the code base with it.

**Comment by ScorpioCPH** [comment]
@BenTheElder Thanks for your kindly suggestion!

Addressed your comments, PTAL.

**Comment by ScorpioCPH** [comment]
@BenTheElder Ping :-)

**Comment by krzyzacy** [comment]
ok
/hold cancel

**Pull request merged**